src/clients/mattermost.ts

```diff
@@ -0,0 +1,147 @@
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public getUser(userId: string): Promise<User> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.UserPath]);
+        return axios.get(replace(url, Routes.PV.Identifier, userId), {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public getChannel(channelId: string): Promise<Channel> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.ChannelPath]);
+        return axios.get(replace(url, Routes.PV.Identifier, channelId), {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public createDirectChannel(ids: string[]): Promise<any> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.ChannelDirectPath]);
+        return axios.post(url, ids, {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public getChannelMembers(channelId: string): Promise<ChannelMember[]> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.ChannelPath, Routes.MM.MembersPath]);
+        return axios.get(replace(url, Routes.PV.Identifier, channelId), {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public getUsersById(usersIDs: string[]): Promise<User[]> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.UsersIdPath]);
+        return axios.post(url, usersIDs, {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+        }).then((response: AxiosResponse<any>) => response.data);
+    }
+
+    public getFileUploaded(fileID: string): Promise<any> {
+        const url = routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.FilePath]);
+
+        return axios.get(replace(url, Routes.PV.Identifier, fileID), {
+            headers: {
+                Authorization: `Bearer ${this.config.accessToken}`,
+            },
+            responseType: 'stream',
+        }).then((response) => response.data);
+    }
+
+    public getConfigClient(): Promise<ClientConfig> {
+        const url = new URL(routesJoin([this.config.mattermostUrl, Routes.MM.ApiVersionV4, Routes.MM.ConfigClientPath]));
+        url.searchParams.append('format', 'old');
+
+        return axios.get(url.href).then((response: AxiosResponse<ClientConfig>) => response.data);
```

Comment:
Currently this is making an API request as an unauthenticated user which does not returns the expected output. 

Suggested change:

```
        return axios.get(url.href, {
            headers: {
                Authorization: `Bearer ${this.config.accessToken}`,
            },
        }).then((response: AxiosResponse<ClientConfig>) => response.data);
  ```

Reply:
```suggestion
        return axios.get(url.href, {
            headers: {
                Authorization: `Bearer ${this.config.accessToken}`,
            },
        }).then((response: AxiosResponse<ClientConfig>) => response.data);
```